packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -0,0 +1,482 @@
+
+  /// The height of the gap between [children] in the vertical
+  /// "overflow" layout.
+  ///
+  /// This parameter is only used if the horizontal layout overflows, i.e.
+  /// if there isn't enough horizontal room for the [children] and [spacing].
+  ///
+  /// Defaults to 0.0.
+  ///
+  /// See also:
+  ///
+  ///  * [spacing], The width of the gap between each pair of children
+  ///    for the default horizontal layout.
+  final double overflowSpacing;
+
+  /// The horizontal alignment of the [children] within the vertical
+  /// "overflow" layout.
+  ///
+  /// This parameter is only used if the horizontal layout overflows, i.e.
+  /// if there isn't enough horizontal room for the [children] and [spacing].
+  /// In that case the overflow bar will expand to fill the available
+  /// width and it will layout its [children] in a column. The
+  /// horizontal alignment of each child within that column is
+  /// defined by this parameter and the [textDirection]. If the
+  /// [textDirection] is [TextDirection.ltr] then each child will be
+  /// aligned with the left edge of the available space for
+  /// [OverflowBarAlignment.start], with the right edge of the
+  /// available space for [OverflowBarAlignment.end]. Similarly, if the
+  /// [textDirection] is [TextDirection.rtl] then each child will
+  /// be aligned with the right edge of the available space for
+  /// [OverflowBarAlignment.start], and with the left edge of the
+  /// available space for [OverflowBarAlignment.end]. For
+  /// [OverflowBarAlignment.center] each child is horizontally
+  /// centered with the available space.
+  ///
+  /// Defaults to [OverflowBarAlignment.start].
+  ///
+  /// See also:
+  ///
+  ///  * [overflowDirection], which defines the order that the
+  ///    [OverflowBar]'s children appear in, if the horizontal layout
+  ///    overflows.
+  final OverflowBarAlignment overflowAlignment;
+
+  /// Defines the order that the [children] appear in, if
+  /// the horizontal layout overflows.
+  ///
+  /// This parameter is only used if the horizontal layout overflows, i.e.
+  /// if there isn't enough horizontal room for the [children] and [spacing].
+  ///
+  /// If the children do not fit into a single row, then they
+  /// are arranged in a column. The first child is at the top of the
+  /// column if this property is set to [VerticalDirection.down], since it
+  /// "starts" at the top and "ends" at the bottom. On the other hand,
+  /// the first child will be at the bottom of the column if this
+  /// property is set to [VerticalDirection.up], since it "starts" at the
+  /// bottom and "ends" at the top.
+  ///
+  /// Defaults to [VerticalDirection.down].
+  ///
+  /// See also:
+  ///
+  ///  * [overflowAlignment], which defines the horizontal alignment
+  ///    of the children within the vertical "overflow" layout.
+  final VerticalDirection overflowDirection;
+
+  /// Determines the order that the [children] appear in for the default
```

Comment:
If this is null, then it uses the inherited text direction right? Probably worth mentioning that in the documentation.

Reply:
👍